### PR TITLE
(DEV-3529) [Prayer Module] Creates MyPrayer List

### PR DESCRIPTION
**Pull request by cdanwards**
## DESCRIPTION

As a User, I want to be able to see my previous prayer requests so that I can see God's faithfulness in meeting prayers.

### What does this PR do, or why is it needed?

This PR creates the `MyPrayerCard` component that is the basic block displaying the prayers and also the `MyPrayers` component, which displays the prayers as a list.

### How do I test this PR?

Currently, the `MyPrayers` is housed on the Connect tab. Make sure you have a few prayers in Alpha-Rock and also be sure that they're public. (Should be moved to prayer Module once other components have been built)

---

> I am affirming this is my _best_ work ([Ecclesiastes 9:10](https://www.bible.com/bible/97/ECC.9.10.MSG))

## TODO

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] Closes [DEV-3683](https://newspring.atlassian.net/secure/RapidBoard.jspa?rapidView=71&modal=detail&selectedIssue=DEV-3683) and [DEV-3684](https://newspring.atlassian.net/secure/RapidBoard.jspa?rapidView=71&modal=detail&selectedIssue=DEV-3684)
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android if applicable
- [x] Set two relevant reviewers

## REVIEW

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

---

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Review by redreceipt — APPROVED** [review]

**Review by redreceipt — DISMISSED** [review]

File: packages/newspringchurchapp/src/tabs/connect/Connect.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ import {
   Icon,
   PaddedView,
 } from '@apollosproject/ui-kit';
+import MyPrayers from '../../ui/MyPrayer';
```
Comment (IsaacHardy) [inline packages/newspringchurchapp/src/tabs/connect/Connect.js]: I'm not 100% excited about the naming here.

File: packages/newspringchurchapp/src/ui/MyPrayer/MyPrayerCard.tests.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  id: 'PrayerRequest:66a4d75b02b447556e4e3806430a9946',
+  firstName: 'Aaron',
+  lastName: 'Attendee',
+  text:
+    'Aaron is the man with the plan who owns a can and drives to Japan which has a lot of sand.',
+  enteredDateTime: '2017-04-10T15:14:54.013',
+};
+
+describe('the MyPrayerCard component', () => {
+  it('renders Prayer1', () => {
```
Comment (IsaacHardy) [inline packages/newspringchurchapp/src/ui/MyPrayer/MyPrayerCard.tests.js]: The other tests do not really differ. I don't exactly see a reason to create tests for prayer2 - prayer5

File: packages/newspringchurchapp/src/ui/MyPrayer/MyPrayerCard.tests.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+
+import Providers from 'newspringchurchapp/src/Providers';
+
+import MyPrayerCard from './MyPrayerCard';
+
+const prayer1 = {
```
Comment (IsaacHardy) [inline packages/newspringchurchapp/src/ui/MyPrayer/MyPrayerCard.tests.js]: If all of these were in one `const` set to an array or prayers, you would only need one test.

File: packages/newspringchurchapp/src/tabs/connect/Connect.js (hunk 1 above)
Comment (IsaacHardy) [reply packages/newspringchurchapp/src/tabs/connect/Connect.js]: Using `my` is what I don't think is necessary.

File: packages/newspringchurchapp/src/ui/MyPrayer/index.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+const MyPrayers = () => (
+  <MyPrayersView>
+    <Header>
+      <H5>My Prayers</H5>
+    </Header>
+    <Query query={getMyPrayers} fetchPolicy="cache-and-network">
+      {({ data: { getCurrentPersonPrayerRequests = [] } = {} }) => (
+        <Content>
+          {getCurrentPersonPrayerRequests
+            .slice(0)
```
Comment (IsaacHardy) [inline packages/newspringchurchapp/src/ui/MyPrayer/index.js]: Why was this needed?

File: packages/apollos-data-prayer/src/prayer-requests/schema.js [hunk 5]
```diff
@@ -24,6 +24,7 @@ const prayerRequestSchema = gql`
     firstName: String
     lastName: String
     text: String!
+    enteredDateTime: String
```
Comment (IsaacHardy) [inline packages/apollos-data-prayer/src/prayer-requests/schema.js]: Is this a required property?

**Review by IsaacHardy — CHANGES_REQUESTED** [review]
Have questions

File: packages/newspringchurchapp/src/tabs/connect/Connect.js (hunk 1 above)
Comment (cdanwards) [reply packages/newspringchurchapp/src/tabs/connect/Connect.js]: I'm going to change it to `UserPrayers`. I think this is more descriptive of the intent.

File: packages/newspringchurchapp/src/tabs/connect/Connect.js [hunk 6]
```diff
@@ -15,6 +15,7 @@ import {
   Icon,
   PaddedView,
 } from '@apollosproject/ui-kit';
+import UserPrayers from '../../ui/UserPrayer';
```
Comment (IsaacHardy) [inline packages/newspringchurchapp/src/tabs/connect/Connect.js]: I believe the UserPrayerCard is a great name, however, this component represents the entire view of user prayers so I don't think just UserPrayer is descriptive enough. I'd recommend something like UserPrayerList.

**Review by IsaacHardy — CHANGES_REQUESTED** [review]

**Review by IsaacHardy — APPROVED** [review]
Boom!

**Review by redreceipt — APPROVED** [review]

**Comment by IsaacHardy** [comment]
@cdanwards I'm merging now but don't hesitate if two reviewers "okay" the PR.
